content/articles/api-authentication-with-django-knox-and-postman-testing/index.md

```diff
@@ -0,0 +1,343 @@
+
+We will be importing the APIViews made in `api.py` as the views for the routes. Here is the code for the routing.
+
+```python
+from django.urls import path, include
+from .api import SignUpAPI, SignInAPI, MainUser
+from knox import views as knox_views
+
+urlpatterns = [
+    path('api/auth/', include('knox.urls')),
+    path('api/auth/register', SignUpAPI.as_view()),
+    path('api/auth/login', SignInAPI.as_view()),
+    path('api/auth/user', MainUser.as_view()),
+    path('api/auth/logout',knox_views.LogoutView.as_view(), name="knox-logout"),
+]
+```
+
+Note that we are including `knox.urls`, this is to route every HTTP calls with the endpoints `api/auth` to match the secured API. 
+
+And also we are using the built-in `LogoutView` for logout of users if requested.
+
+### Postman Testing
+
+If you already have a postman, go ahead and test the API following the procedures below. But if you don't have it installed on your local machine [Download Postman](/https://www.postman.com/downloads/).
+
+Remember that the server has to be running before you can make HTTP calls to the endpoints. For that reason, we will be starting the server with the command below.
+
+```bash
+$ python manage.py runserver
+```
+
+Before we can reference the API we have to register and then login which I assume you know.
+
+1. Make a `GET` request to `http://localhost:8080/api/leads` and `http://localhost:8080/api/user`. Since the user is not logged in, then a validation error should be sent.
+"fcdec16877f60c59f8552533444d1e809ffc4b7436606ad94139f82ad25542d1"
+
+2. Now make a `POST` request to  `http://localhost:8080/api/auth/register` and under the body, choose `raw` >> `JSON`. Provide the JSON object for the user having `username, `email`, and `password`.
+
+The response will come with a token that will be used to authorize the user when logging in. See image below;
+
+![Postman Registration API Testing](/engineering-education/api-authentication-with-django-knox-and-postman-testing/register.png)
+
+
+3. Copy out the token sent from the registration and paste it in the value section of `Authorisation` under the `Headers` by typing `Token` before it as shown below.
+
+ Now Make a `POST` request to `http://localhost:8080/api/auth/login` with only the `username`, and `password` of the registered user. Having logged in, another fresh token will be sent for log out.
+
+![Postman Log In API Testing](/engineering-education/api-authentication-with-django-knox-and-postman-testing/login.png)
+
+4. Finally, make a `GET` request to the leads API endpoints and you will be allowed to access it. This must be done before the token expired.
+
+#### Conclusion
+
+In this tutorial, we covered the implementation of Django-Knox, and we were able to demonstrate the token wise of the framework by building the rest API endpoints that are based on user authentication and authorization.
```

Comment:
Change to " In this tutorial, we covered the implementation of Django-Knox. We demonstrated the token wise of the framework by building the rest of the API endpoints based on user authentication and authorization."